pkg/cmd/infra/router/router.go

```diff
@@ -83,6 +95,48 @@ func (o *RouterSelection) RouteSelectionFunc() controller.RouteHostFunc {
 	}
 }
 
+func (o *RouterSelection) AdmissionCheck(route *routeapi.Route) error {
+	if len(route.Spec.Host) < 1 {
+		return nil
+	}
+
+	if hostInDomainList(route.Spec.Host, o.BlacklistedDomains) {
+		glog.V(4).Infof("host %s in list of denied domains", route.Spec.Host)
+		return fmt.Errorf("host in list of denied domains")
+	}
+
+	if o.WhitelistedDomains.Len() > 0 {
+		glog.V(4).Infof("Checking if host %s is in the list of allowed domains", route.Spec.Host)
+		if hostInDomainList(route.Spec.Host, o.WhitelistedDomains) {
+			glog.V(4).Infof("host %s admitted - in the list of allowed domains", route.Spec.Host)
+			return nil
+		}
+
+		glog.V(4).Infof("host %s rejected - not in the list of allowed domains", route.Spec.Host)
+		return fmt.Errorf("host not in the allowed list of domains")
+	}
+
+	glog.V(4).Infof("host %s admitted", route.Spec.Host)
+	return nil
+}
+
+// RouteAdmissionFunc returns a func that checks if a route can be admitted
+// based on blacklist & whitelist checks and wildcard routes policy setting.
+// Note: The blacklist settings trumps the whitelist ones.
+func (o *RouterSelection) RouteAdmissionFunc() controller.RouteAdmissionFunc {
+	return func(route *routeapi.Route) (error, bool) {
+		if err := o.AdmissionCheck(route); err != nil {
+			return err, false
+		}
+
+		if len(route.Spec.WildcardPolicy) > 0 && route.Spec.WildcardPolicy != routeapi.WildcardPolicyNone && !o.AllowWildcardRoutes {
```

Comment:
Well it could be but it felt better to separate them out as the admission check is an outright rejection - in/not in black/white lists. But a wildcard policy check in here (even if the router policy doesn't allow it), we still want to handle the host ... this would make it behave similar to previous/olders routers (where wildcards are not allowed - well would ignore the api field if the api server was updated but not the router and the router would process the route host). I'll make the other changes.